subprojects/docs/src/docs/userguide/troubleshooting.adoc

```diff
@@ -15,29 +15,169 @@
+
+If you get "permission denied", that means that Gradle likely exists in the correct place, but it is not executable. You can fix this using `chmod +x path/to/executable` on *nix-based systems.
+
+==== Other installation failures
+
+If `gradle --version` works, but all of your builds fail with the same error, it is possible there is a problem with one of your Gradle build configuration scripts.
+
+You can verify the problem is with Gradle scripts by running `gradle help` which executes configuration scripts, but no Gradle tasks. If the error persists, build configuration is problematic.
+If not, then the problem exists within the execution of one or more of the requested tasks (Gradle executes configuration scripts first, and then executes build steps).
+
+
+=== Debugging dependency resolution
+
+Common dependency resolution issues such as resolving version conflicts are covered in <<troubleshooting_dependency_resolution>>.
+
+You can see a dependency tree and see which resolved dependency versions differed from what was requested by clicking the _Dependencies_ view and using the search functionality, specifying the resolution reason.
+
+++++
+<figure>
+    <title>Debugging dependency conflicts with build scans</title>
+    <imageobject>
+        <imagedata fileref="img/troubleshooting-dependency-management-build-scan.png"/>
+    </imageobject>
+</figure>
+++++
+
+The link:https://scans.gradle.com/s/sample/troubleshooting-userguide/dependencies?expandAll&filters=WzFd&toggled=W1swXSxbMF0sWzAsMF0sWzAsMV1d[actual build scan] with filtering criteria is available for exploration.
+
+=== Troubleshooting slow Gradle builds
+
+For build performance issues (including “slow sync time”), see the guide to link:https://guides.gradle.org/performance/[Improving the Performance of Gradle Builds].
+
+Android developers should watch a presentation by the Android SDK Tools team about link:https://youtu.be/7ll-rkLCtyk[Speeding Up Your Android Gradle Builds]. Many tips are also covered in the Android Studio user guide link:https://developer.android.com/studio/build/optimize-your-build.html[on optimizing build speed].
+
+
+=== Debugging build logic
+
+==== Attaching a debugger to your build
+
+You can set breakpoints and debug <<sec:packaging_a_plugin,buildSrc and standalone plugins>> in your Gradle build itself by setting the `org.gradle.debug` property to “true” and then attaching a remote debugger to port 5005.
+
+----
+❯ gradle help -Dorg.gradle.debug=true --no-daemon
+----
+
+In addition, if you’ve adopted the Kotlin DSL, you can also debug build scripts themselves.
+
+The following video demonstrates how to debug an example build using IntelliJ IDEA.
+
+++++
+<figure>
+    <title>Interactive debugging of a build script</title>
+    <imageobject>
+        <imagedata fileref="img/remote-debug-gradle.gif"/>
+    </imageobject>
+</figure>
+++++
+
+**NOTE:** You must either stop running Gradle Daemons or run with `--no-daemon` when using debug mode.
```

Comment:
We should probably use the AsciiDoc notation here to align with other content formatting.